### DISCUSSION TITLE
What is the purpose of LocalFileImageDataProvider?

### DISCUSSION THREAD

**Discussion by yccheok** [onevcat/Kingfisher#1782] [Q&A] [answered]
Hi,

All my images are located in `applicationSupportDirectory` local disk.

I believe Kingfisher would able to help me to achieve a smooth scrolling experience, when my `UICollectionView` is displaying tone of images. I am still not able to validate my assumption, as my implementation is work in progress...

I studied https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet#imagedataprovider

I notice that `LocalFileImageDataProvider` is used, if the image is loaded from local disk

```
let provider = LocalFileImageDataProvider(fileURL: attachment.url)
imageView.kf.setImage(with: provider)
```

However, I tested without using `LocalFileImageDataProvider`, with the following code

`imageView.kf.setImage(with: attachment.url)`

Image was able to show in `UIImageView` without issue.

If it is so, may I know what is the purpose of LocalFileImageDataProvider? Thanks.

**Comment by onevcat** [comment, marked as answer]
The `LocalFileImageDataProvider` is used inside `kf.setImage` when you give it a file URL, so it is just provide you a shortcut. In your case, I guess you can just ignore the `LocalFileImageDataProvider` unless you need [another loading queue](https://github.com/onevcat/Kingfisher/blob/70ef0ba088a26561bfee1a2a15efc157696ce33e/Sources/General/ImageSource/ImageDataProvider.swift#L88) for some reason.

**Reply by yccheok** [reply]
Hi @onevcat 

Thank you very much for your valuable input.

May I know, is it necessary to use Kingfisher to load image from local disk and display in `UIImageView`?

As, I have read some community comments, on using image loading framework is redundant for local disk file.

I am very new in iOS development. 

My previous experience in Android told me that, using image loading framework is necessary, to avoid UI thread from doing I/O operations, to ensure the smoothness of the app.

I am not quite sure whether such assumption holds true still, for iOS?

My use case is, I will display a lot of images (from disk files) in `UICollectionView`. I guess without Kingfisher, user will experience a very "laggy" scrolling experience. I am still not able to validate such assumption, as I haven't reach to that implementation point.

May I know, based on your experience, is Kingfisher required for such use case?

Thanks.

**Reply by onevcat** [reply]
> My use case is, I will display a lot of images (from disk files) in UICollectionView. I guess without Kingfisher, user will experience a very "laggy" scrolling experience.

It really depends on how many images and how large of them you need to display. In most cases for normal UI, it should be fine. But there is also no harm to use Kingfisher for it. There are a few buiit-in optimization in Kingfisher even for loading images from a file URL. So I suggest you use it as well.

**Reply by yccheok** [reply]
Thank you for your suggestion.
